include/aspect/stokes_matrix_free.h

```diff
@@ -392,6 +455,69 @@ namespace aspect
       static
       void declare_parameters (ParameterHandler &prm);
 
+      /**
+       * Return a reference to the DoFHandler that is used for velocity in
+       * the block GMG solver.
+       */
+      const DoFHandler<dim> &
+      get_dof_handler_v () const override;
```

Comment:
I am late, but in the current form you add a lot of very specific functions to SimulatorAccess. For all other cases we usually expose the owning object through `SimulatorAccess` (e.g. `SimulatorAccess::get_newton_handler`) and let  the returned object decide what to expose. Could we do the same here: `SimulatorAccess::get_stokes_matrix_free_handler`? It saves us some functions in SimulatorAccess and reduces the number of places we would have to change if something changes about the implementation.

Reply:
Yes, I see that now. I was able to remove all other matrix-free functions in SimulatorAccess with the exception of `is_stokes_matrix_free()`.